src/main/java/org/apache/sysml/api/python/SystemML.py

```diff
@@ -247,3 +257,163 @@ def getStringRDD(self, varName, format):
         #    return rdd
         #except Py4JJavaError:
         #    traceback.print_exc()
+
+def getNumCols(numPyArr):
+	if len(numPyArr.shape) == 1:
+		return 1
+	else:
+		return numPyArr.shape[1]
+       
+def convertToJavaMatrix(sc, src):
+	if isinstance(src, np.ndarray):
+		from array import array
+		numCols = getNumCols(src)
+		numRows = src.shape[0]
+		if src.dtype.type is np.float64:
+			arr = src.reshape(-1)
+		else:
+			arr = array('d', src.reshape(-1))
+		buf = bytearray(arr.tostring())
+		return sc._jvm.org.apache.sysml.runtime.instructions.spark.utils.RDDConverterUtilsExt.convertPy4JArrayToMB(buf, numRows, numCols)
+	else:
+		raise Exception('Type is not supported')
```

Comment:
You can do `src = np.asarray(src)` in the beginning to make sure data structures like list of lists etc are supported.